logstash-core/lib/logstash/runner.rb

```diff
@@ -115,6 +115,11 @@ class LogStash::Runner < Clamp::StrictCommand
     :attribute_name => "http.port",
     :default => LogStash::SETTINGS.get_default("http.port"), &:to_i
 
+  option ["--http.environment"], "HTTP_ENVIRONMENT",
```

Comment:
I don't think this flag is something the user should care about, end of the day we should be setting the hole environment (for all logstash) or nothing, this makes other moving part of logstash in strange status.

I think is better to have this variable injected somehow when we run test, or either be setup in regular mode when run from command line. Does it makes sense?

happy to provide more clarity.